native/lambda_game_engine/src/game.rs

```diff
@@ -525,10 +530,18 @@ fn apply_zone_effects(players: &mut HashMap<u64, Player>, zone: &Zone) {
         });
 
         outside_players.into_iter().for_each(|player| {
-            player.apply_effects_if_not_present(
-                &current_modification.outside_radius_effects,
-                EntityOwner::Zone,
-            );
+            if player.status == PlayerStatus::Alive {
```

Comment:
Is this if necessary? We weren't using it before. I think you can just use the condition where the player died after this update (just leave the next if instead of these two).

Reply:
It's necessary because we want to push to the killfeed only in the case in which the player died because of the effect of the zone. So we need to check if it's alive first and only push to the killfeed if the player was killed right then. Otherwise, we would be pushing to the killfeed a lot of times unnecessarily.

Reply:
I just pushed a comment to clarify this logic